### PR TITLE
Folder Structure + Make File

### PR DESCRIPTION
**Pull request by mohamedali5**
- Merged multiple makefiles into 1 single makefile and modified the commands within to be able to clean, run tests, run simulator, or run all.
- Modified the project folder structure to meet the coding practices agreed upon in class.
- Modified README file to accommodate the changes to the folder structure.
- Added cadmium and DESTimes to lib folder.

### CODE REVIEW

File: README.txt [hunk 1]
```diff
@@ -8,45 +8,74 @@ README.txt
...
-	receiverCadmium.hpp
-	senderCadmium.hpp
-	subnetCadmium.hpp
+bin [This folder contains all .exe files generated for the project]
+	test [This folder contains all .exe files generated for testing only]
+		RECEIVER_TEST.exe
+		SENDER_TEST.exe
+		SUBNET_TEST.exe
+	top_model [This folder contains all .exe files generated for simulator only]
+	ABP.exe
```
Comment (p-sky) [inline README.txt]: ABP.exe should be indented (move right one tab or 4 spaces)

Comment (p-sky) [inline README.txt]: ABP.exe should be indented (move right one tab or 4 spaces)

File: README.txt [hunk 2]
```diff
@@ -58,30 +87,28 @@ some data for testing and some simulator outputs]
...
-    
-2 - Run the unit tests
-	2.1. Run subnet test
-		1 - Open the terminal. Press in your keyboard Ctrl+Alt+t
-		2 - Set the command prompt in the test/subnet folder. To do so, type in the terminal the path to this folder.
-			Example: cd Documents/AlternateBitProtocol/test/subnet
-		3 - To compile the test, type in the terminal:
-			make clean; make all
-		4 - To run the test, type in the terminal "./NAME_OF_THE_COMPILED_FILE". For this specific test you need to type:
-			./SUBNET
```
Comment (p-sky) [inline README.txt:70]: Should be ./SUBNET_TEST

File: README.txt [hunk 3]
```diff
@@ -8,45 +8,74 @@ README.txt
...
-	senderCadmium.hpp
-	subnetCadmium.hpp
+bin [This folder contains all .exe files generated for the project]
+	test [This folder contains all .exe files generated for testing only]
+		RECEIVER_TEST.exe
+		SENDER_TEST.exe
+		SUBNET_TEST.exe
+	top_model [This folder contains all .exe files generated for simulator only]
+	ABP.exe
+build [This folde contains all .o files generated for the project]
```
Comment (p-sky) [inline README.txt]: The word "folde" is missing 'r' letter

**Review by p-sky — CHANGES_REQUESTED** [review]
Please address the 3 comments raised

File: README.txt [hunk 4]
```diff
@@ -58,30 +87,28 @@ some data for testing and some simulator outputs]
...
+		./bin/top_model/ABP data/top_model/input_abp_1.txt
+	5 - To check the output of the simulation, open  "abp_output.txt" present in data/top_model folder.
 	6 - To execute the simulator with different inputs
 		6.1. Create new .txt files with the same structure as input_abp_0.txt or input_abp_1.txt
 		6.2. Run the simulator using the instructions in step 4
 		6.3. If you want to keep the output, rename abp_output.txt. To do so, type in the terminal: "mv abp_output.txt NEW_NAME"
-			Example: mv abp_output.txt abp_output_0.txt
+			Example: mv data/top_model/abp_output.txt data/top_model/abp_output_0.txt
+	7 - To run the subnet test, type in the terminal "./NAME_OF_THE_COMPILED_FILE". For this specific test you need to type:
+			./bin/test/SUBNET
```
Comment (p-sky) [inline README.txt]: Should be ./bin/test/SUBNET_TEST

**Review by p-sky — CHANGES_REQUESTED** [review]

File: README.txt (hunk 1 above)
Comment (pcsmartapps) [reply README.txt]: Addressed

File: README.txt (hunk 3 above)
Comment (pcsmartapps) [reply README.txt]: DONE

File: README.txt (hunk 4 above)
Comment (pcsmartapps) [reply README.txt]: Done

File: README.txt (hunk 2 above)
Comment (pcsmartapps) [reply README.txt:70]: This is in a red color and it refers to the old readme file.

File: README.txt (hunk 1 above)
Comment (syedomarazam) [reply README.txt]: fixed it. please review the latest README file

File: README.txt [hunk 5]
```diff
@@ -58,30 +87,28 @@ some data for testing and some simulator outputs]
...
+		./bin/top_model/ABP data/top_model/input_abp_1.txt
+	5 - To check the output of the simulation, open  "abp_output.txt" present in data/top_model folder.
 	6 - To execute the simulator with different inputs
 		6.1. Create new .txt files with the same structure as input_abp_0.txt or input_abp_1.txt
 		6.2. Run the simulator using the instructions in step 4
 		6.3. If you want to keep the output, rename abp_output.txt. To do so, type in the terminal: "mv abp_output.txt NEW_NAME"
-			Example: mv abp_output.txt abp_output_0.txt
+			Example: mv data/top_model/abp_output.txt data/top_model/abp_output_0.txt
+	7 - To run the subnet test, type in the terminal "./NAME_OF_THE_COMPILED_FILE". For this specific test you need to type:
+			./bin/test/SUBNET_TEST
```
Comment (shubhamagrawal6629) [inline README.txt]: Remove tab or 4 spaces as it should be in line with 7.1

Comment (shubhamagrawal6629) [inline README.txt]: Not in alignment with 7.1 . I think you need to remove tab or 4 spaces

Comment (syedomarazam) [reply README.txt]: resolved the indentation issue

**Review by shubhamagrawal6629 — APPROVED** [review]
Everything works fine and the folder structure looks good

**Review by syedomarazam — APPROVED** [review]
Fixed all the comments received for Task#1

**Review by p-sky — APPROVED** [review]

**Review by pcsmartapps — APPROVED** [review]

**Pull request merged**